keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1004 @@
+ClusterRoleBindings.  It is possible to grant specific users the ability
+read and/or update all TrustAnchorSets that correspond to a specific signer.
+
+To enable consumption by workloads, the new `trustAnchorSets` Kubelet projected
+volume source allows projecting a trust anchor set from a particular signer into
+the container filesystem, with the contents of the projected files updating as
+the corresponding TrustAnchorSets are updated.
+
+### User Stories
+
+#### Trust Anchor Distribution for Private CAs
+
+In my cluster, I operate a certificates/v1 signer implementation called
+`example.com/server-tls`.  This is backed by a private CA hierarchy.  By some
+unspecified mechanism, I issue certificates from this CA hierarchy to a server
+workload (a pod named `server/server`).  My client workload (a pod named
+`client/client`) needs access to the trust anchors for the CA in order to
+connect to the server.  The client should be able to continually establish
+connections to the server without downtime, even as I rotate the trust anchors
+that back the CA.
+
+To do this, I update the controller that implements `example.com/server-tls` to
+create a TrustAnchorSets object like this one:
+```yaml
+apiVersion: v1alpha1
+kind: TrustAnchorSets
+metadata:
+  name: example-com-server-tls-abcdefg
+spec:
+  signerName: example.com/server-tls
+status:
+  sets:
+  - name: default
+    pemTrustAnchors:
+    - "<... PEM DATA ...>"
+```
+
+As the set of trust anchors for my CA changes due to rotations and revocations,
+my controller updates the status of the TrustAnchorSets object.
+
+I then update my client pod to add a `trustAnchorSets` volume:
+```diff
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: client
+  name: client
+spec:
+  containers:
+    name: main
+    image: my-image
+    volumeMounts:
++    - mountPath: /var/run/example-com-server-tls-trust-anchors
++      name: example-com-server-tls-trust-anchors
++      readOnly: true
+  volumes:
++  - name: example-com-server-tls-trust-anchors
++    projected:
++      sources:
++      - trustAnchorSets:
++          signerName: example.com/server-tls
++          sets:
++          - singleFile:
++              set: default
++              path: ca_certificates.pem
+```
+
+When my client pod is scheduled onto a node, Kubelet writes the chosen trust
+anchor set into the container filesystem of my main container.  My main
+container doesn't start until `ca_certificates.pem` is present and non-empty.
+
+I ensure that my client application is capable of reading a set of trust anchors
+from a file containing PEM-wrapped DER-formatted X.509 certificates at the
+location I've specified
+(`/var/run/example-com-server-tls-trust-anchors/ca_certificates.pem`).
+
+I ensure that my client application notices changes to `ca_certificates.pem`
+(using inotify or periodic polling), and properly updates its TLS stack when
+that happens.
+
+#### TrustAnchorSets for Webhook Backends
+
+(To be added)
```

Comment:
Aggregated API server section?

Reply:
I added a stub.  I think we might need to cut the actual ability to use ClusterTrustBundles for webhooks and aggregated API servers if we want to hit 1.25